### PR TITLE
Fix deserialize

### PR DESCRIPTION
**Pull request by Keith-CY**
1. use custom `padStart` for duktape
2. fix header slice on deserializing dynvec and union
3. stop deserialize on empty value received

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -214,7 +215,7 @@ class Molecule {
       case 'option':
         return deserializeOption(value)
       case 'union':
-        return deserializeUnion(value, [littleHexToInt(value.slice(0, HEADER_ELEMENT_SIZE))])
+        return deserializeUnion(value, [littleHexToInt(value.slice(0, HEADER_ELEMENT_SIZE * 2))])
```
Comment (Keith-CY) [inline src/index.ts]: @duanyytop the second param derives from the first, so is it possible to remove the second param?

Comment (duanyytop) [reply src/index.ts]: Yes. We can remove the second param

Comment (duanyytop) [reply src/index.ts]: I update it in this pr or you update ?

Comment (Keith-CY) [reply src/index.ts]: > I update it in this pr or you update ?

Please submit a PR to this branch.

Comment (duanyytop) [reply src/index.ts]: OK

Comment (Keith-CY) [reply src/index.ts]: > OK

And I found another problem and working on it now, I'll set this PR to HOLD until I fix it.

Comment (Keith-CY) [reply src/index.ts]: > > OK
> 
> And I found another problem and working on it now, I'll set this PR to HOLD until I fix it.

😂 It's a wrong test case so the code is OK

Comment (Keith-CY) [reply src/index.ts]: This update from `HEADER_ELEMENT_SIZE` to `HEADER_ELEMENT_SIZE * 2` is wrong according to the README, it's a 32-bit number, please fix the update in your PR

Comment (duanyytop) [reply src/index.ts]: OK. I think we should check the index of union serialized data and items length. I will update this logic.

Comment (Keith-CY) [reply src/index.ts]: > OK. I think we should check the index of union serialized data and items length. I will update this logic.

Merged

**Pull request merged**